bears/general/AnnotationBear.py

```diff
@@ -5,6 +5,41 @@
 from coalib.results.SourceRange import SourceRange
 from coalib.results.AbsolutePosition import AbsolutePosition
 from coala_utils.string_processing.Core import unescaped_search_for
+from coala_utils.decorators import (enforce_signature, generate_repr,
+                                    generate_ordering)
+
+
+@generate_repr(('id', hex),
+               'origin',
+               'string_ranges',
+               'comment_ranges',
+               'exception_msg',
+               'message')
+@generate_ordering('string_ranges',
+                   'comment_ranges',
+                   'exception_msg',
+                   'contents',
+                   'message',
+                   'origin',
+                   'message_base')
+class AnnotationResult(HiddenResult):
+
+    @enforce_signature
+    def __init__(self, origin,
+                 string_ranges: (tuple, None)=None,
+                 comment_ranges: (tuple, None)=None,
+                 exception_msg: (str, None)=None):
+
+        Result.__init__(self, origin,
+                        exception_msg if exception_msg is not None else '')
+
+        self.contents = (dict(strings=string_ranges,
+                              comments=comment_ranges)
+                         if exception_msg is None
+                         else exception_msg)
+        self.string_ranges = string_ranges
```

Comment:
Ok, With the new AnnotationBear we are sending precise singleline/multiline string/comment info. I'm encapsulating the return value into class. So, we will have to modify this PR. putting it on halt for a while.

Reply:
@aptrishu , how about this patch doesnt add `self.string_ranges` or `self.comment_ranges`, and gets merged, and then your patch improves `AnnotationResult` by adding any new fields that the consumer needs?

Reply:
`self.string_ranges` and `self.comment_ranges` has to be there. This pr is ready, I thought we would merge its modified version after the new annotationbear. but yeah it is fine. we can do the improvements later.